### ISSUE TITLE
Needs better error handling when WP install folder does not have acceptable permissions

### ISSUE THREAD

**Issue by pippinsplugins** [wp-cli/wp-cli#4594]
When the folder WordPress is installed in has restrictive file permissions (no read nor write), WP CLI fails with a non-helpful error message.
```
pippin@pippin-XPS-13-9360:~/sites/wp$ wp plugin list
Error: This does not seem to be a WordPress install.
Pass --path=`path/to/wordpress` or run `wp core download`.
```
The error gives the impression that the `wp` command is being run from the wrong folder, when in reality it's the correct folder that just lacks the proper permissions / ownership for the user running the command.

It would save a lot of debugging time and trouble if a meaningful error message was returned.

**Comment by schlessera** [comment]
Here's the current check to detect a WP installation:
```php
private function wp_exists() {
	return is_readable( ABSPATH . 'wp-includes/version.php' );
}
```

We could check "existence" and "readability" separately, and adapt the error message accordingly.

**Comment by schlessera** [comment]
Fixed via #4626
